shark_turbine/kernel/wave/register_analysis.py

```diff
@@ -33,16 +42,43 @@ def set_register_shape(trace: CapturedTrace, custom: CustomOp) -> None:
                     idx
                 ]
             )
-            set_register_shape(trace, iter_arg)
+            set_register_shape(trace, iter_arg, vector_map)
             custom.fx_node.thread_shape = iter_arg.fx_node.thread_shape
             break
+        elif isinstance(custom_user, ReduceOp):
+            get_thread_shape = lambda reg_shape: max(reg_shape)
```

Comment:
Do you need this lambda? Seems like it is just computing the max?

Reply:
Done! Initially, I wrapped it around lambda to be more verbose on what we are trying to do with the max